### PR TITLE
App 59/android logs

### PR DESCRIPTION
**Pull request by atar9500**

### CODE REVIEW

**Comment by DanielZiv-SF** [comment]
now that we have the ability to write logs, let's also collect errors logs. can you please go over the code and find places where we haven't wrapped in try catch?

**Comment by DanielZiv-SF** [comment]
+ can you please upload a log example of normal usage of the app? (regular mfa from background + foreground)

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        Log.wtf(tag, "");
+    }
+
+    private enum LogLevel {
+        d("LOG"),
+        e("ERROR"),
+        i("LOG"),
+        v("LOG"),
+        w("WARN"),
+        wtf("WTF");
```
Comment (DanielZiv-SF) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java:180]: wtf?

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        }
+        return mLogger;
+    }
+
+    private LoggerWrapper(Context context) {
+        File externalDirectory = context.getExternalFilesDir(null);
+        String baseUrl = externalDirectory.getAbsolutePath();
+        mLogsFilesUrl = baseUrl + "/silverfort/logs";
+        mLogFileUrl = mLogsFilesUrl + "/logs.txt";
+        mLogFileUrl2 = mLogsFilesUrl + "/logs_2.txt";
```
Comment (DanielZiv-SF) [inline lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java]: what is logs 2?

**Review by DanielZiv-SF — CHANGES_REQUESTED** [review]
how we avoid overwriting the same file by the java process and the react native proces if they try to write at the same time?
not sure that the best approach is to open and close the file on every log
isn't there a package of android logger that we can add an handler for the log file instead of implementing it on our own?

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java (hunk 1 above)
Comment (atar9500) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java:180]: on android there's a log level called wtf for extreme cases

File: lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java (hunk 2 above)
Comment (atar9500) [reply lib/android/app/src/main/java/com/wix/reactnativenotifications/utils/LoggerWrapper.java]: our second file for file rotation

**Comment by atar9500** [comment]
> how we avoid overwriting the same file by the java process and the react native proces if they try to write at the same time?
> not sure that the best approach is to open and close the file on every log
> isn't there a package of android logger that we can add an handler for the log file instead of implementing it on our own?

Basically the file writings are both occurring on the JAVA side, but probably not on the same process (RN process basically fires an event to the JAVA process with the files we want to write).
I think finding a logger library can be good, but a bit overkill for our solution (making sure file is not opened at the same time we want to modify it) rather than attaching an observer to the file ( https://developer.android.com/reference/android/os/FileObserver#top_of_page ) to make sure we write to file when it is possible only.

**Comment by atar9500** [comment]
> now that we have the ability to write logs, let's also collect errors logs. can you please go over the code and find places where we haven't wrapped in try catch?

[logs.txt](https://github.com/welldone-software/react-native-notifications/files/5792423/logs.txt)

**Review by DanielZiv-SF — APPROVED** [review]

**Pull request merged**